app/assets/javascripts/discourse/app/components/glimmer-site-header.gjs

```diff
@@ -243,7 +243,7 @@ export default class GlimmerSiteHeader extends Component {
 
   @bind
   dockCheck() {
-    if (this._docAt === null) {
+    if (this._docAt === undefined || this._docAt === null) {
```

Comment:
FWIW `Boolean(0)` is also false, so that would also let through the `0` value. So I think the linter is correct in this case. I wish there were some kind of 'nullish' check built in to JS 😢